torch/_inductor/ir.py

```diff
@@ -6593,6 +6593,80 @@ def _has_aliased_buffers(buffers: Sequence[IRNode]) -> bool:
     return len(OrderedSet(id(buffer) for buffer in buffers)) < len(buffers)
 
 
+@ir_dataclass(frozen=False)
+class InvokeSubgraph(ExternKernel):
+    subgraph: Optional[Subgraph] = None
+    operands: Optional[List[TensorBox]] = None
+    outputs: Optional[List[MultiOutput]] = None
+
+    def __init__(
+        self, subgraph: Subgraph, operands: List[TensorBox], layout: MultiOutputLayout
+    ):
+        super().__init__(
+            name=None,
+            layout=layout,
+            inputs=operands,
+        )
+        self.subgraph = subgraph
+        self.name = V.graph.register_buffer(self)
+        V.graph.register_operation(self)
+
+    @classmethod
+    def create(cls, subgraph: Subgraph, operands):
+        fx_operands = V.graph.current_node.args[-1]
+        fake_operands = [x.meta["val"] for x in fx_operands]  # type: ignore[union-attr]
+
+        # Realize the inputs. Also intermediates can have different strides than
+        # the inputs of the subgraph. So, force the intermediates to have same
+        # strides as that of subgraph inputs.
+        operands = [cls.realize_input(x) for x in operands]
+        # TODO(anijain2305) - Support symint inputs in future.
+        fake_strides = [fake_operand.stride() for fake_operand in fake_operands]
```

Comment:
we expect sym expr inside inductor, not symints.. it's a bit confusing, something we should fix. 

Can you convert the symints first:

https://github.com/pytorch/pytorch/blob/c272526ea54b8063c596addb48cd3a87ab91a5ef/torch/_inductor/graph.py#L1490-L1493

Also -`require_exact_strides` doesn't work great with aliases/NonOwningLayout.... this should mostly work for now though.